src/Data/AffineSpace.hs

```diff
@@ -122,3 +140,58 @@ instance (AffineSpace p) => AffineSpace (a -> p) where
   type Diff (a -> p) = a -> Diff p
   (.-.)              = liftA2 (.-.)
   (.+^)              = liftA2 (.+^)
+
+
+
+newtype GenericDiff p = GenericDiff (Diff (Gnrx.Rep p ()))
```

Comment:
Well, the `Rep` is really a low-level plumbing thing. In particular for large record types, it becomes considerably branched and confusing in case it's ever unwrapped in a type error. A `type` can't prevent this, but a `newtype` can.